### PR TITLE
feat(intellij): add setting to specify non-root nx workspace path

### PR DESCRIPTION
**Pull request by MaxKless**
- move to settings folder
- register workspace path project setting
- rework setting rendering
- apply workspace path setting on startup

### CODE REVIEW

File: apps/intellij/src/main/kotlin/dev/nx/console/nxls/WorkspacePathSetting.kt [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    private val inputField = JBTextField()
+    override fun render(panel: Panel) {
+        panel.apply {
+            row {
+                label("Workspace path")
+                cell(inputField)
+                    .horizontalAlign(HorizontalAlign.FILL)
+                    .comment(
+                        "Set this if your Nx workspace is not at the root of the project opened in IntelliJ",
```
Comment (Cammisuli) [inline apps/intellij/src/main/kotlin/dev/nx/console/nxls/WorkspacePathSetting.kt:19]: Dont mention "IntelliJ". Nx Console can be installed on: AppCode, Aqua, CLion, GoLand, IntelliJ, PhpStorm, PyCharm, Rider, RubyMine, AND WebStorm 😅

File: apps/intellij/src/main/kotlin/dev/nx/console/nxls/WorkspacePathSetting.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import com.intellij.ui.components.JBTextField
+import com.intellij.ui.dsl.builder.MAX_LINE_LENGTH_WORD_WRAP
+import com.intellij.ui.dsl.builder.Panel
+import com.intellij.ui.dsl.gridLayout.HorizontalAlign
+import dev.nx.console.settings.NxConsoleSettingBase
+
+class WorkspacePathSetting : NxConsoleSettingBase<String?> {
+
+    private val inputField = JBTextField()
```
Comment (iguissouma) [inline apps/intellij/src/main/kotlin/dev/nx/console/nxls/WorkspacePathSetting.kt:11]: Maybe change to a Field with Browse Button

Comment (MaxKless) [reply apps/intellij/src/main/kotlin/dev/nx/console/nxls/WorkspacePathSetting.kt:11]: Yes this is planned (why the PR is only draft for now)

**Comment by Cammisuli** [comment]
For awareness - generate doesnt work anymore because NxExecutor is still using the base path of the project. 

Here's the proposed solution, extend `com.intellij.openapi.project.Project` to have `nxBasePath` like so:
```
/**
 * Get the base path of the current Nx project. Will get the settings first, then default to the
 * `basePath`
 */
val Project.nxBasePath: String
    get() =
        NxConsoleProjectSettingsProvider.getInstance(this).workspacePath
            ?: basePath ?: throw IllegalStateException("Base path is not found for project")
```

Then we can use `project.nxBasePath` instead of `project.basePath`

**Review by Cammisuli — CHANGES_REQUESTED** [review]
see previous comment

File: apps/intellij/src/main/kotlin/dev/nx/console/generate/run_generator/RunGeneratorManager.kt [hunk 3]
```diff
@@ -57,15 +58,15 @@ class RunGeneratorManager(val project: Project) {
                     {
                         val nxExecutable =
                             NxExecutable.getExecutablePath(
-                                project.basePath
+                                project.nxBasePath
                                     ?: throw Exception("Project base path does not exist")
```
Comment (Cammisuli) [inline apps/intellij/src/main/kotlin/dev/nx/console/generate/run_generator/RunGeneratorManager.kt:62]:
```suggestion
```

File: apps/intellij/src/main/kotlin/dev/nx/console/generate/run_generator/RunGeneratorManager.kt [hunk 4]
```diff
@@ -78,7 +79,7 @@ class RunGeneratorManager(val project: Project) {
                         val consoleBuilder =
                             TextConsoleBuilderFactory.getInstance().createBuilder(project)
                         val console = consoleBuilder.console
-                        project.basePath?.let {
+                        project.nxBasePath?.let {
```
Comment (Cammisuli) [inline apps/intellij/src/main/kotlin/dev/nx/console/generate/run_generator/RunGeneratorManager.kt:82]:
```suggestion
                        project.nxBasePath.let {
```

**Review by Cammisuli — APPROVED** [review]
Just a couple of nitpicks, but approved overall 🙂

**Pull request merged**